### DISCUSSION TITLE
Disk Usage feature

### DISCUSSION THREAD

**Discussion by Mulugruntz** [geldata/gel#2454] [Ideas]
I don't think I've seen this proposition anywhere yet.

A feature to find out the disk usage of each table, index and relation would be very useful (probably for many people and not just me).

Right now, my database is taking an unreasonable amount of disk space and I cannot analyze it in any way.

PG of course has such tools https://wiki.postgresql.org/wiki/Disk_Usage but EdgeDB could have something similar, albeit with a more "modern" grammar.

Maybe directly in the introspection it could make sense?

```
WITH MODULE schema 
SELECT ObjectType { 
    id, 
    name, 
    indexes: {
        id,
        name,
        expr,
        size: {  # new stuff here
            element_approx,
            total_size,
        }
    }
    size: {  # new stuff here
        element_approx,
        total_size,
    }
};
```
Out:
```json
{
  "id": "442c6fcd-9905-11eb-9d2e-e9a9c0200988",
  "indexes": [
    {
      "expr": ".timestamp",
      "id": "443176ca-9905-11eb-bbb9-27c7b76d3404",
      "name": "timestamp_96de3c58",
      "size": {
         "element_approx": 42420,
         "total_size": 12563299
       }
    }
  ],
  "size": {
    "element_approx": 42420,
    "total_size": 8144640
  },
  "name": "default::History"
}
```

And same for any other stuff that could make sense.

**Comment by elprans** [comment]
Agree, this would be a useful feature.  We might want to include index usage stats as well.
